Cannot find module './Lazy.js' from 'cases.js'

# Bug Report

<!--
  Please fill in each section completely. Thank you!
-->

### 💻 Code

<!-- Please post the relevant code sample here as well-->
<!-- How to reproduce it -->
```ts
// index.ts
import { delay, pipe, chunk, map, toAsync, toArray } from "@fxts/core";

const executeWithLimit = <T>(
  funcs: IterableIterator<() => Promise<T>>,
  limit: number
): Promise<T[]> => {
  return pipe(
    funcs,
    chunk(limit),
    map((_funcs) => _funcs.map((func) => func())),
    map((promises) => Promise.all(promises)),
    toAsync,
    toArray
  ).then((arr) => arr.flat());
};

type File = {
  name: string;
  body: string;
  size: number;
};

async function getFile(name: string, size = 1000): Promise<File> {
  console.log(`${name} 시작`);
  return delay(size, { name, body: "...", size });
}

async function test() {
  const names = [
    "1-img.png",
    "2-img.png",
    "3-img.png",
    "4-img.png",
    "5-img.png",
    "6-img.png",
  ];

  const funcs = names.map((name) => {
    return () => getFile(name);
  });

  const files: File[] = await executeWithLimit(funcs[Symbol.iterator](), 3);

  console.log(files);
}

(async () => {
  await test();
})();
```

### 🙁 Actual behavior

<!-- What happened, and why it was wrong -->
When running the code with `bun run index.ts`, a module resolution error occurs internally within @fxts/core. It seems like Bun is strictly checking for file extensions in ESM imports, and it fails to resolve './Lazy.js' requested by 'dist/esm/cases.js'.

```
error: Cannot find module './Lazy.js' from '/workspace/node_modules/@fxts/core/dist/esm/cases.js'
````

### 🙂 Expected behavior

<!-- What you expected to happen instead, and why -->
run index.ts without error

---

### Version Information

<!-- Please let me know your environment -->

- Operating system: macOS 15.7.1
- Typescript: ^5
- Nodejs: 20.19.5
- Bun: 1.2.22

--- a/src/cases.ts
+++ b/src/cases.ts
@@ -1,4 +1,4 @@
-import { chunk } from "./Lazy";
+import chunk from "./Lazy/chunk";
 
 type Predicate<T> = (value: T) => boolean;
 type Refinement<T, T1 extends T> = (value: T) => value is T1;
@@ -187,9 +187,7 @@ function cases<
  * ```
  */
 function cases<T, R>(...fns: ((x: T) => boolean | R)[]) {
-  return function (
-    value: T,
-  ): typeof fns extends EvenArray ? R | typeof value : R {
+  return (value: T): typeof fns extends EvenArray ? R | typeof value : R => {
     for (const pair of chunk(2, fns) as Generator<
       [Predicate<T>, Mapper<T, R>] | [DefaultFn<T, R>, undefined]
     >) {

